pkg/roachprod/roachprod.go

```diff
@@ -2387,6 +2388,71 @@ func CreateLoadBalancer(
 	return nil
 }
 
+// Deploy deploys a new version of cockroach to the given cluster. It currently
+// does not support clusters running external SQL instances.
+// TODO(herko): Add support for virtual clusters (external SQL processes)
+func Deploy(
+	ctx context.Context,
+	l *logger.Logger,
+	clusterName, applicationName, version string,
+	pauseDuration time.Duration,
+	sig int,
+	wait bool,
+	maxWait int,
+	secure bool,
+) error {
+	supportedApplicationNames := []string{"cockroach", "release", "customized"}
+	if !slices.Contains(supportedApplicationNames, applicationName) {
+		return errors.Errorf("unsupported application name %s, supported names are %v", applicationName, supportedApplicationNames)
```

Comment:
Should we just delegate this validation to `Stage`?

Reply:
`stage` already validates this, but unfortunately it will allow `workload`, whereas we only want the `cockroach` variants.

What I can do is maybe negate the validation on the command side and disallow `workload`, but then again if someone adds a new 'application type' that will need to be updated (without necessarily even knowing it exists), versus if you want something new here you can add it.

Reply:
Oh I see, that makes sense. I forgot about `workload`. Maybe add a note here explaining the reason?